deeppavlov/core/data/data_fitting_iterator.py

```diff
@@ -0,0 +1,68 @@
+"""
+Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from random import Random
+from typing import List, Generator, Tuple, Any
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.common.log import get_logger
+
+logger = get_logger(__name__)
+
+
+@register('data_fitting_iterator')
+class DataFittingIterator:
+    """
+    Dataset iterator for fitting models, e. g. vocabs, kNN, vectorizers.
+    Data is passed as a list of strings(documents).
+    Generate batches (for large datasets).
+    """
+
+    def __init__(self, data: List[str], doc_ids: List[Any]=None,
+                 seed: int = None, shuffle: bool = None, batch_size: int = None,
+                 *args, **kwargs) -> None:
+
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.random = Random(seed)
+        self.data = data
+        self.doc_index = doc_ids or self.get_doc_ids()
+        self.data = data
+
+    def get_doc_ids(self):
+        ids = [i for i in range(len(self.data))]
+        return ids
+
+    def gen_batch(self, batch_size=1000, shuffle=False) -> Generator[Tuple[List[list],
+                                                                           List[int]], Any, None]:
+        _batch_size = self.batch_size or batch_size
+        _shuffle = self.shuffle or shuffle
```

Comment:
Argument to the method should have priority. Also, since `shuffle` is boolean, it should be compared with proper `None` checks

Reply:
About priority - my prev comment.
About booleans. It would be an issue, when `self.shuffle=True`, but in this particular function a user wants `shuffle=False`. This would be impossible, because `_shuffle = True or False` will lead to `True` anyway. However, I don't see this would be a proper use case for this class. If users want `shuffle=False`, they should init a class with proper values. So I'd better raise a warning here, instead of allowing users mess with functionality as they please.

Reply:
Same point as in the previous comment.